src/v/storage/segment_utils.h

```diff
@@ -238,6 +238,13 @@ is_compactible_control_batch(const model::record_batch_type batch_type) {
            || batch_type == model::record_batch_type::group_commit_tx;
 }
 
+inline bool
+compactible_offset_translator_batch(const model::record_batch_type batch_type) {
+    // certain batches not contributing to the kafka offset space are
+    // compactible
+    return batch_type == model::record_batch_type::datalake_translation_state;
+}
+
```

Comment:
i think this will lead to an issue in which we may result in an offset inconsistency. Offset translator state is not checkpointed to kvstore on every update, if those batches are compacted they won't contribute to the offset translator state anymore leading to Kafka offsets shift.

Reply:
good catch, I think it would also mess up partition moves after compaction as the new node builds OTS based on the compacted log. will remove this commit.

Reply:
naive question: is there a way to catch this with testing in the future? This ought to be a scenario we can unit test in theory.

Reply:
https://github.com/redpanda-data/redpanda/pull/23817